xdg-desktop-portal-luminous: init at 0.1.8

Rishik-Y opened the pull request:
## Package Information

- Package name: xdg-desktop-portal-luminous
- Latest released version: 0.1.8
- Current version on the unstable channel: none
- Current version on the stable/release channel: none

An alternative to xdg-desktop-portal-wlr for wlroots compositors. This project is a stand alone binary and does not depend on grim. libwayshot is used as the screencopy backend to enable screenshots.

<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

Rishik-Y committed:
xdg-desktop-portal-luminous: init at 0.1.8

HeitorAugustoLN commented on pkgs/by-name/xd/xdg-desktop-portal-luminous/package.nix:
```diff
@@ -0,0 +1,71 @@
+
+  LIBCLANG_PATH = "${libclang.lib}/lib";
+
+  configurePhase = ''
+    meson build --prefix=$out --libexecdir=$out/lib --buildtype=release
+  '';
+
+  buildPhase = ''
+    ninja -C build
+  '';
+
+  installPhase = ''
+    ninja -C build install
+  '';
```
This isn't needed, `meson` and `ninja`  already set up hooks to do this

Rishik-Y replied:
Yes, i initially indeed didnt keep it assuming it wouldnt be needed 
however for some reason i needed to explicitely state this for the package to successfully build,
So i would suggest to keep it unless have some other solution

HeitorAugustoLN replied:
```suggestion
  configurePhase = ''
    runHook preConfigure
    
    meson build --prefix=$out --libdir=$out/lib --libexecdir=$out/libexec --buildtype=release
    
    runHook postConfigure
  '';
  
  ninjaFlags = [
    "-C"
    "build"
  ];
```

HeitorAugustoLN commented on pkgs/by-name/xd/xdg-desktop-portal-luminous/package.nix:
```diff
@@ -0,0 +1,71 @@
+    ninja -C build install
+  '';
+
+  meta = with lib; {
+    description = "xdg-desktop-portal backend for wlroots based compositors, providing screenshot and screencast";
+    homepage = "https://github.com/waycrate/xdg-desktop-portal-luminous";
+    license = licenses.gpl3;
+    platforms = platforms.linux;
+    maintainers = [ ];
+  };
```
`with lib` in `meta` is discouraged and `lib.licenses.gpl3` is deprecated.

```suggestion
  meta = {
    description = "xdg-desktop-portal backend for wlroots based compositors, providing screenshot and screencast";
    homepage = "https://github.com/waycrate/xdg-desktop-portal-luminous";
    license = lib.licenses.gpl3Only;
    platforms = lib.platforms.linux;
    maintainers = [ ];
  };
```

HeitorAugustoLN commented on pkgs/by-name/xd/xdg-desktop-portal-luminous/package.nix:
```diff
@@ -0,0 +1,71 @@
+    ninja -C build install
+  '';
+
+  meta = with lib; {
+    description = "xdg-desktop-portal backend for wlroots based compositors, providing screenshot and screencast";
+    homepage = "https://github.com/waycrate/xdg-desktop-portal-luminous";
+    license = licenses.gpl3;
+    platforms = platforms.linux;
+    maintainers = [ ];
+  };
```
Since you are adding a new package, please add yourself as a maintainer for the package

Rishik-Y replied:
Hello 👋
I actually did keep that initially but it was showing me failed to build while not detecting Maintainers
Do i still keep my username anyway?

Rishik-Y replied:
Sorry my bad will add myself in it.

HeitorAugustoLN commented on pkgs/by-name/xd/xdg-desktop-portal-luminous/package.nix:
```diff
@@ -0,0 +1,71 @@
+
+  src = fetchFromGitHub {
+    owner = "waycrate";
+    repo = pname;
```
`repo = pname` is an antipattern, it is better to be hardcoded instead

```suggestion
    repo = "xdg-desktop-portal-luminous";
```

HeitorAugustoLN commented on pkgs/by-name/xd/xdg-desktop-portal-luminous/package.nix:
```diff
@@ -0,0 +1,71 @@
+  src = fetchFromGitHub {
+    owner = "waycrate";
+    repo = pname;
+    rev = "v${version}";
```
Can use `tag` instead

```suggestion
    tag = "v${version}";
```

HeitorAugustoLN commented on pkgs/by-name/xd/xdg-desktop-portal-luminous/package.nix:
```diff
@@ -0,0 +1,71 @@
+  pipewire,
+}:
+
+rustPlatform.buildRustPackage rec {
```
Not strictly necessary, but could use `finalAttrs` pattern instead for easier overriding

```suggestion
rustPlatform.buildRustPackage (finalAttrs: {
```

HeitorAugustoLN commented on pkgs/by-name/xd/xdg-desktop-portal-luminous/package.nix:
```diff
@@ -0,0 +1,71 @@
+    libxkbcommon
+  ];
+
+  LIBCLANG_PATH = "${libclang.lib}/lib";
```
```suggestion
  env.LIBCLANG_PATH = "${lib.getLib libclang}/lib";
```

HeitorAugustoLN commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 389933`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>xdg-desktop-portal-luminous</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>xdg-desktop-portal-luminous</li>
  </ul>
</details>

SuperSandro2000 commented on pkgs/by-name/xd/xdg-desktop-portal-luminous/package.nix:
```diff
@@ -0,0 +1,72 @@
+  configurePhase = ''
+    runHook preConfigure
+
+    meson build --prefix=$out --libdir=$out/lib --libexecdir=$out/libexec --buildtype=release
```
Is the hook not working here? https://github.com/NixOS/nixpkgs/blob/master/pkgs/by-name/me/meson/setup-hook.sh

HeitorAugustoLN replied:
For some reason, it is not being run in the configure phase. Not sure why

Rishik-Y replied:
any update on this?

SuperSandro2000 replied:
because https://github.com/NixOS/nixpkgs/blob/master/pkgs/build-support/rust/build-rust-package/default.nix#L172-L176 disables https://github.com/NixOS/nixpkgs/blob/master/pkgs/by-name/me/meson/setup-hook.sh#L91

K900 replied:
`buildRustPackage` is the wrong tool here, if you want Meson to be driving the build.

SuperSandro2000 replied:
So switch it to stdenv.mkDerivation then

K900 commented on pkgs/by-name/xd/xdg-desktop-portal-luminous/package.nix:
```diff
@@ -0,0 +1,62 @@
+  pipewire,
+}:
+
+rustPlatform.buildRustPackage (finalAttrs: {
```
Still using buildRustPackage when you shouldn't be.

K900 replied:
That package is also wrong.

Rishik-Y replied:
i see. Thanks for informing.

Rishik-Y commented:
Hi @K900,
Could you confirm it now?

Rishik-Y commented:
Hi @K900, 
Sorry for disturbing you multiple times.
I have modified the code.
Hope the code is fine now.
I have also modified [xdg-desktop-portal-shana](https://github.com/NixOS/nixpkgs/pull/393432) as well,
So if you could take a quick look at that, i would appreciate it.
Thanks

HeitorAugustoLN commented on pkgs/by-name/xd/xdg-desktop-portal-luminous/package.nix:
```diff
@@ -0,0 +1,69 @@
+    repo = "xdg-desktop-portal-luminous";
+    tag = "0.1.8";
+    hash = "sha256-7i6+arKWubziDmy07FocDDiJdOWAszhO7yOOI1iPfds=";
+  };
```
just use `inherit (finalAttrs) pname version src;`

K900 commented on pkgs/by-name/xd/xdg-desktop-portal-luminous/package.nix:
```diff
@@ -0,0 +1,68 @@
+    libxkbcommon
+  ];
+
+  env.LIBCLANG_PATH = "${lib.getLib libclang}/lib";
```
You can use `rustPlatform.bindgenHook` instead of this. If you do that, you can also remove clang and libclang from your buildInputs.

Rishik-Y replied:
Yep Done! :+1:

K900 commented on pkgs/by-name/xd/xdg-desktop-portal-luminous/package.nix:
```diff
@@ -0,0 +1,67 @@
+    rustc
+    cargo
+    clang
+    libclang
```
libclang and clang should not be necessary now.

Rishik-Y replied:
*sigh* I am so sorry, for making silly mistakes.
I have updated the code again

K900 reviewed (approved):
LGTM now.

wineee commented on pkgs/by-name/xd/xdg-desktop-portal-luminous/package.nix:
```diff
@@ -0,0 +1,65 @@
+  rustc,
+  cargo,
+  libclang,
+  clang,
```
libclang and clang not used

Rishik-Y replied:
My bad, Fixed it!
